internal/credential/service_list_credentials.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package credential
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/pagination"
+)
+
+// CredentialService defines the interface expected
+// to get the total number of credentials and deleted ids.
+type CredentialService interface {
+	EstimatedCredentialCount(context.Context) (int, error)
+	ListDeletedCredentialIds(context.Context, time.Time) ([]string, time.Time, error)
+	ListCredentials(context.Context, string, ...Option) ([]Static, error)
+}
```

Comment:
The credential service is implemented by any credential repos, in this case the only one is the static credential repo.